rustls/src/client/handy.rs

```diff
@@ -101,7 +102,7 @@ impl client::ClientSessionStore for ClientSessionMemoryCache {
 
     fn set_tls12_session(
         &self,
-        _server_name: &ServerName,
+        _server_name: ServerName<'static>,
```

Comment:
nit: this is carried over from the old version, but I think `_server_name` should be `server_name` since it's used on L111 for the `get_or_insert_default_and_edit` call.

If you agree & don't mind, the same feedback applies to `tls_session`, & `remove_tls12_session`

Reply:
I think this is intentional -- you may have glossed over the `#[cfg(feature = "tls12")]` guarding the `get_or_insert_default_and_edit()` call, which means that this would only be conditionally used dependending on Cargo features, which in turn means it would generate warnings in the other case. So going to leave this as is.

Reply:
Ahh! yes, my mistake.